src/screens/IdentityPin.js

```diff
@@ -0,0 +1,215 @@
+// GNU General Public License for more details.
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+'use strict';
+
+import React, { Component, useState } from 'react';
+import { StyleSheet } from 'react-native';
+import { withNavigation } from 'react-navigation';
+import colors from '../colors';
+import Background from '../components/Background';
+import ButtonMainAction from '../components/ButtonMainAction';
+import TextInput from '../components/TextInput';
+import KeyboardScrollView from '../components/KeyboardScrollView';
+import { withAccountStore } from '../util/HOC';
+import testIDs from '../../e2e/testIDs';
+import ScreenHeading from '../components/ScreenHeading';
+import fontStyles from '../fontStyles';
+
+export default withAccountStore(withNavigation(IdentityPin));
+
+function IdentityPin({ navigation, accounts }) {
+	const initialState = {
+		confirmation: '',
+		focusConfirmation: false,
+		pin: '',
+		pinMismatch: false,
+		pinTooShort: false
+	};
+	const [state, setState] = useState(initialState);
+	const updateState = delta => setState({ ...state, ...delta });
+
+	const submit = async () => {
+		const isIdentityCreation = navigation.getParam('isNew');
+		const { pin, confirmation } = state;
+		if (pin.length >= 6 && pin === confirmation) {
+			if (isIdentityCreation) {
+				const resolve = navigation.getParam('resolve');
+				setState(initialState);
+				resolve(pin);
+			} else {
+				// await accounts.save(accounts.getSelectedKey(), account, pin);
+				// const resetAction = StackActions.reset({
+				// 	actions: [
+				// 		NavigationActions.navigate({ routeName: 'LegacyAccountList' }),
+				// 		NavigationActions.navigate({ routeName: 'AccountDetails' })
+				// 	],
+				// 	index: 1, // FIXME workaround for now, use SwitchNavigator later: https://github.com/react-navigation/react-navigation/issues/1127#issuecomment-295841343
+				// 	key: undefined
+				// });
+				// navigation.dispatch(resetAction);
+			}
+		} else {
+			if (pin.length < 6) {
+				updateState({ pinTooShort: true });
+			} else if (pin !== confirmation) updateState({ pinMismatch: true });
+		}
+	};
+
+	const testPin = async () => {
+		const { pin } = state;
+		if (pin.length >= 6) {
+			try {
+				const seed = await accounts.unlockIdentitySeed(pin);
+				const resolve = navigation.getParam('resolve');
+				setState(initialState);
+				resolve(seed);
+			} catch (e) {
+				updateState({ pin: '', pinMismatch: true });
+				//TODO record error times;
+			}
+		} else {
+			updateState({ pinTooShort: true });
+		}
+	};
+
+	const showHintOrError = () => {
+		if (state.pinTooShort) {
+			return ' Your pin must be at least 6 digits long!';
+		} else if (state.pinMismatch) {
+			return " Pin codes don't match!";
+		}
+		return ' Choose a PIN code with 6 or more digits';
+	};
+
+	const onPinInputChange = (stateName, pinInput) => {
+		if (/^\d+$|^$/.test(pinInput)) {
```

Comment:
small suggestion: the `/^\d+$|^$/.test` could be put as an export from `util/numbers`

Reply:
create a `utils/regex.js` file to order the regexs